common/src/hand_off.rs

```diff
@@ -377,8 +378,12 @@ impl FirmwareHandoffTable {
     /// Perform valdity check of the table's data.
     /// The fields below should have been populated by ROM with
     /// valid data before it transfers control to mutable code.
+    /// This function can only be called from non test case environment
+    /// as this function accesses the registers to get the reset_reason.
     pub fn is_valid(&self) -> bool {
```

Comment:
We can pass reset reason as an argument and keep the FHT hardware agnostic.

Reply:
I had it that way but then I realized that is_valid is also getting called from try_load, which again gets called from some other tests. So one way or other we will end up putting this limitation. If we keep the reset_reason local to this function we just need modification in one test case. This is a small thing and we can revisit this later, if needed.